src/dump.c

```diff
@@ -614,12 +661,27 @@ static void jl_serialize_value_(jl_serializer_state *s, jl_value_t *v, int as_li
     else if (jl_is_expr(v)) {
         jl_expr_t *e = (jl_expr_t*)v;
         size_t l = jl_array_len(e->args);
+        if (e->head == call_sym) {
+            if (l == 2) {
+                write_uint8(s->s, TAG_CALL1);
+                jl_serialize_value(s, jl_exprarg(e, 0));
+                jl_serialize_value(s, jl_exprarg(e, 1));
+                return;
+            }
+            else if (l == 3) {
+                write_uint8(s->s, TAG_CALL2);
+                jl_serialize_value(s, jl_exprarg(e, 0));
+                jl_serialize_value(s, jl_exprarg(e, 1));
+                jl_serialize_value(s, jl_exprarg(e, 2));
+                return;
+            }
```

Comment:
Do we need `TAG_CALL0` and `TAG_INVOKE{0...2}` also?

Reply:
We could, but I found call1 and call2 covered about 85% of cases.